### PR TITLE
Prepare for react-lite → React switch

### PR DESCRIPTION
**Pull request by martinpitt**
This picks a bunch of commits from PR #9263 which are safe and fine for react-lite as well, and removes some conflicts to current master. Let's land this separately so that it's easier to eventually get in the complete switch.

### CODE REVIEW

**Comment by martinpitt** [comment]
These are all @mareklibra 's changes, and I approve them. I can't formally do it here, as it's nominally my PR. I manually tested terminal (including resizing), selectors, the ReactDOM.render affected pages etc., and it all works just fine.

File: test/verify/check-ovirt [hunk 1]
```diff
@@ -83,6 +83,19 @@ class TestOVirtMachines(MachineCase):
         self.hack_for_missing_vdsm()
         self.pass_install_dialog()
 
+    # mimics pressing keys
+    def _setVal(self, selector, value):
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply test/verify/check-ovirt]: Done, and I added a separate commit to robustify `set_input_text()` like Marek did in the `_setVal` helper function.

File: test/common/testlib.py [hunk 2]
```diff
@@ -239,6 +239,7 @@ def set_input_text(self, selector, val):
         self.set_val(selector, "")
         self.focus(selector)
         self.key_press(val)
+        b.wait_val(selector, val)
```
Comment (mvollmer) [inline test/common/testlib.py]: `self.wait_val`

Comment (martinpitt) [reply test/common/testlib.py]: Argh, fixed.

**Comment by martinpitt** [comment]
Two storage test failures on ubuntu-1604: https://fedorapeople.org/groups/cockpit/logs/pull-9938-20180831-235353-9c47b50d-verify-ubuntu-1604/log.html#201

Retrying for comparison.

**Comment by martinpitt** [comment]
Actually @mvollmer already approved up there, just wanted the `wait_val` typo fixed (which broke tests). ubuntu-1604 udisks bugs are unrelated, he's working on them. So this can land now IMHO.

**Pull request merged**